lib/seek/fair_data_station/writer.rb

```diff
@@ -33,7 +53,14 @@ def reset_data_file_cache
         @data_file_cache = {}
       end
 
-      def build_assay(datastation_assay, contributor, policy, projects, sample, study)
+      def preload_data_file_cache(data_files)
```

Comment:
question: why do you need to do this?

Reply:
data files can be connected in multiple places, so may be encountered more than once. The cache is used to preven't multiple duplicates of the data files being created. For the import this is built up whilst as they are created, but for the update the cache needs preloading, otherwise new ones will start being created again.